packages/react-dom/src/shared/HTMLDOMPropertyConfig.js

```diff
@@ -83,34 +83,6 @@ var HTMLDOMPropertyConfig = {
     htmlFor: 'for',
     httpEquiv: 'http-equiv',
   },
-  DOMMutationMethods: {
-    value: function(node, value) {
-      if (value == null) {
-        return node.removeAttribute('value');
```

Comment:
This would only happen when switching from a controlled to uncontrolled component right? Since we're not removing the attribute anymore could this break anybody?

Reply:
This is for elements that use the `value` attribute, like `<progress />` and `<meter />`. I am not sure inputs can actually ever have a null `value` prop because:

1. [An input sets an "initial value" on creation](https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOMFiberInput.js#L81)
2. If that value is null, it falls back to `defaultValue`
3. [If defaultValue is null, it falls back to an empty string](https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOMFiberInput.js#L140)

Additionally, React yells at you for switching between a controlled and uncontrolled input, and if the value is ever null.

Reply:
Doesn't react yell if you switch to undefined? Null gets a different warning about using an empty string

Reply:
Yes. You get the following warnings when switching to and from `undefined`:

> Warning: A component is changing an uncontrolled input of type undefined to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components

Reply:
Right, it warns and it will also remove the attribute. With this change, I believe it will warn but the attribute will not be removed.

Reply:
Good catch. It does not remove the attribute, but the behavior is inconsistent with master. In the absence of a `value` attribute, an input reverts back to the initial value it was created with. For example:

**React 16.2.0:**

![value](https://user-images.githubusercontent.com/590904/33353339-5063232c-d47b-11e7-9d8d-202e985636a9.gif)

**However, with this change, the attribute is never reverted to that initial value:**

![value-local](https://user-images.githubusercontent.com/590904/33353440-d70ef446-d47b-11e7-835a-1c0887a3aea8.gif)

I'll add test coverage for this case. Sorry to neglect the distinction between `null` and `undefined` 😦